frontend/src/components/pages/secrets/update-secret-modal.tsx

```diff
@@ -102,6 +114,39 @@ export const UpdateSecretModal = ({ isOpen, onClose, secretId }: UpdateSecretMod
                   {(field) => <field.PasswordField label="Value" data-testid="secret-value-field" />}
                 </form.AppField>
 
+                <form.AppField name="scopes">
+                  {({ state, handleChange, handleBlur }) => (
+                    <FormField label="Scopes" errorText=" " isInvalid={state.meta.errors?.length > 0}>
+                      <Select
+                        placeholder="Select scopes"
+                        onChange={(nextValue) => {
+                          if (isMultiValue(nextValue) && nextValue) {
+                            handleChange(nextValue.map(({ value }) => value));
+                          }
+                        }}
+                        options={scopeOptions}
+                        defaultValue={scopeOptions.filter((so) => matchingSecret?.scopes?.some((s) => so.value === s))}
+                        isMulti
+                        onBlur={handleBlur}
+                      />
```

Comment:
I think this is perfect to create MultiSelect component, just copy this into its own component and we can add it to the list of available form fields.

Reply:
Feel free to grab me or I can contribute to this PR if you like, we can also do some pairing if you need to do knowledge sharing, both work for me.

Reply:
lets create a ticket for it, not sure if we are going to continue with this approach

Reply:
sure, I can clean up as well, let's not block this PR.